includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -1602,8 +1609,16 @@ public function search_products( $term, $type = '', $include_variations = false,
 			$searchand        = '';
 
 			foreach ( $search_terms as $search_term ) {
-				$like              = '%' . $wpdb->esc_like( $search_term ) . '%';
-				$term_group_query .= $wpdb->prepare( " {$searchand} ( ( posts.post_title LIKE %s) OR ( posts.post_excerpt LIKE %s) OR ( posts.post_content LIKE %s ) OR ( wc_product_meta_lookup.sku LIKE %s ) )", $like, $like, $like, $like ); // phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+				$like = '%' . $wpdb->esc_like( $search_term ) . '%';
+
+				// Variations should also search the parent's meta table for fallback fields.
+				if ( $include_variations ) {
+					$variation_query = $wpdb->prepare( ' OR ( wc_product_meta_lookup.sku = "" AND parent_wc_product_meta_lookup.sku LIKE %s ) ', $like );
+				} else {
+					$variation_query = '';
+				}
+
+				$term_group_query .= $wpdb->prepare( " {$searchand} ( ( posts.post_title LIKE %s) OR ( posts.post_excerpt LIKE %s) OR ( posts.post_content LIKE %s ) OR ( wc_product_meta_lookup.sku LIKE %s ) $variation_query)", $like, $like, $like, $like ); // phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
```

Comment:
If the four `%s` in the format were `%1$s` the prepare would need only one `$like` parameter.

Reply:
Given that the `$` needs to be escaped it makes for a kind of ugly prepare statement. I'll also note that trying to make this change made PHPCS _very_ angry. I received:

1. Error `WordPress.DB.PreparedSQLPlaceholders.LikeWildcardsInQuery`: SQL Wildcards for a LIKE query should be passed through a replacement parameter.
2. Warning `WordPress.DB.PreparedSQLPlaceholders.ReplacementsWrongNumber`: Incorrect number of replacements passed to `$wpdb->prepare()` (It doesn't look like it acknowledges they're repeats!)
3. Warning `WordPress.DB.PreparedSQLPlaceholders.UnquotedComplexPlaceholder`: Complex placeholders used for values in the query string `$wpdb->preapre()` will NOT be quoted automagically.

Given all of this I think it might be best to leave it as it is, what do you think?

Reply:
I wasn't able to get get `$wpdb->prepare` to work with numbered arguments when playing with it in the wp-cli. I agree, I think it's probably fine to leave it as you have it, given all the points you've added on top of that.